OSDOCS-11499: ROSA-HCP - Image registry config to customers (#82728)
openshift/openshift-docs: modules/images-editing-image-registry-settings-hcp.adoc

// Module included in the following assemblies:
//
// * openshift_images/image-configuration-hcp.adoc
// * post_installation_configuration/preparing-for-users.adoc

:_mod-docs-content-type: PROCEDURE
[id="images-editing-image-registry-settings-hcp_{context}"]
= Editing image registry settings for {hcp-title}

After verifying that your image registry is enabled, you can update or edit the image registry with the `rosa edit` command.

> [WARNING]

Author: @chenz4027 do we want this warning in the editing module also and if so I will update it with @andreadecorte new suggetsion?

Author: left in both

// Module included in the following assemblies:
//
// * openshift_images/image-configuration-hcp.adoc
// * post_installation_configuration/preparing-for-users.adoc

:_mod-docs-content-type: PROCEDURE
[id="images-editing-image-registry-settings-hcp_{context}"]
= Editing image registry settings for {hcp-title}

> After verifying that your image registry is enabled, you can update or edit the image registry with the `rosa edit` command.

Reviewer: ```suggestion
Image registry config can be changed in an existing cluster with the `rosa edit` command.
```

Author: Sticking with style guide I have changed it to :
You can change the image registry config with the `rosa edit` command.

[WARNING]
====
When the `allowedRegistries` parameter is defined, all registries, are blocked unless explicitly listed. To prevent pod failure, a list of Red Hat registries is automatically whitelisted, as they are required by payload images within your environment. For disconnected clusters, mirror registries should be added by the user itself.
====

> 

Reviewer: I would add also a second warning with text
```
Changing any registry related parameter will trigger a rollout across all machinepools 
(all machinepool nodes will be recreated, following pod draining from each node).
```
This is also visible below in the CLI output, but I think it should have a dedicated warning too.

Reviewer: Ah it is here https://github.com/openshift/openshift-docs/pull/82728/files#diff-d5b3ac5a27787219fdda141baa9da8251b06649a266719af50d9bb717a109f73R13.
Can we have that warning in both sections? (Configuring image .... and Editing image)

 - arn:aws:iam::<AWS_id>:role/<operator_roles_prefix>-openshift-cluster-csi-drivers-ebs-cloud-credent
 - arn:aws:iam::<AWS_id>:role/<operator_roles_prefix>-openshift-cloud-network-config-controller-cloud
Managed Policies:           Yes
State:                      ready
Private:                    No
Delete Protection:          Disabled
Created:                    Oct 01 2030 09:48:52 UTC
User Workload Monitoring:   Enabled
OIDC Endpoint URL:          https://<endpoint> (Managed)
Audit Log Forwarding:       Disabled
External Authentication:    Disabled
Etcd Encryption:            Disabled
Registry Configuration:
 - Allowed Registries:      <allowed_registry> <1>
 - Insecure Registries:     <insecure_registry> <2>
 - Allowed Registries for Import: <3>
    - Domain Name:          <domain_name> <4>
    - Insecure:             true <5>
----
<1> `Allowed Registries`: A comma-separated list of registries for which image pull and push actions are allowed.
<2> `Insecure Registries`: A comma-separated list of registries which do not have a valid TLS certificate or only support HTTP connections.

> <3> `Allowed Registries for Import`: Limits the container image registries from which normal users can import images. The format should be a comma-separated list of 'domainName:insecure'.
> <4> 'domainName': Specifies a domain name for the registry.
> <5> 'insecure': Indicates whether the registry is secure or insecure.

Reviewer 2: Same changes here re: backticks:
```suggestion
<3> `Allowed Registries for Import`: Limits the container image registries from which normal users can import images. The format should be a comma-separated list of `domainName:insecure'`
<4> `domainName`: Specifies a domain name for the registry.
<5> `insecure`: Indicates whether the registry is secure or insecure.
```

Author: thaks @adellape

// Module included in the following assemblies:
//
// * openshift_images/image-configuration-hcp.adoc
// * post_installation_configuration/preparing-for-users.adoc

:_mod-docs-content-type: PROCEDURE
[id="images-editing-image-registry-settings-hcp_{context}"]
= Editing image registry settings for {hcp-title}

You can change the image registry config with the `rosa edit` command.

[WARNING]
====
When the `allowedRegistries` parameter is defined, all registries are blocked unless explicitly listed. To prevent pod failure, a list of Red{nbsp}Hat registries is automatically whitelisted, as they are required by payload images within your environment. The current list consists of `image-registry.openshift-image-registry.svc:5000,quay.io,registry.redhat.io` and it is also visible when running the `rosa describe cluster` command.
====

[NOTE]
====

> You can change any registry-related parameter, which will trigger a rollout across all machine pools; (all machine pool nodes will be recreated, following pod draining from each node).

Reviewer 2: ```suggestion
You can change any registry-related parameter, which will trigger a rollout across all machine pools; all machine pool nodes will be recreated, following pod draining from each node.
```

// Module included in the following assemblies:
//
// * openshift_images/image-configuration-hcp.adoc
// * post_installation_configuration/preparing-for-users.adoc

:_mod-docs-content-type: PROCEDURE
[id="images-editing-image-registry-settings-hcp_{context}"]
= Editing image registry settings for {hcp-title}

You can change the image registry config with the `rosa edit` command.

[WARNING]
====

> When the `allowedRegistries` parameter is defined, all registries, are blocked unless explicitly listed. To prevent pod failure, a list of Red Hat registries is automatically whitelisted, as they are required by payload images within your environment. The current list consists of `image-registry.openshift-image-registry.svc:5000,quay.io,registry.redhat.io` and it is also visible when running `rosa describe cluster`.

Reviewer 2: ```suggestion
When the `allowedRegistries` parameter is defined, all registries are blocked unless explicitly listed. To prevent pod failure, a list of Red{nbsp}Hat registries is automatically whitelisted, as they are required by payload images within your environment. The current list consists of `image-registry.openshift-image-registry.svc:5000,quay.io,registry.redhat.io` and it is also visible when running the `rosa describe cluster` command.
```

====

[NOTE]
====

> You can change any registry related parameter which will trigger a rollout across all machinepools (all machinepool nodes will be recreated, following pod draining from each node).

Reviewer 2: ```suggestion
You can change any registry-related parameter, which will trigger a rollout across all machine pools; all machine pool nodes will be recreated, following pod draining from each node.
```

====

.Procedure

* Update or edit the image registry for the cluster by running the following command:

+
[source,terminal]
----
$ rosa edit cluster --registry-config-insecure-registries <insecure_registries> \
   --registry-config-allowed-registries <allowed_registries> \
   --registry-config-allowed-registries-for-import <registry_name:insecure> \
   --registry-config-additional-trusted-ca <additional_trusted_ca_file>
----
+
.Example output
[source,terminal]
----

> rosa edit cluster -c <cluster_name> --registry-config-allowed-registries <allowed_registries> \
> ? Changing any registry related parameter will trigger a rollout across all machinepools \
> (all machinepool nodes will be recreated, following pod draining from each node). \
> Do you want to proceed? Yes \ <1>
> I: Updated cluster '<cluster_name>'

Reviewer 2: I don't think we need to show the command again in the "Example output" (since we just showed the command above). We typically want the command and the output in separate code blocks.

Also you don't need the `\` line breaks here (since it's not intended to be actually copy/pasted into a terminal). And there's no matching `<1>` callout, so remove it from the block.

```suggestion
? Changing any registry related parameter will trigger a rollout across all machinepools 
(all machinepool nodes will be recreated, following pod draining from each node). 
Do you want to proceed? Yes
I: Updated cluster '<cluster_name>'
```

Author: removed command from output and applied other changes, thanks

----

.Verification
* Run the `rosa describe` command again, to see if the changes you made to your image registry updated by running the following command:
+
[source,terminal]
----
$ rosa describe cluster --cluster=<cluster_name>
----
+
.Example output
[source,terminal]
----
Name:                       rosa-hcp-test
Domain Prefix:              rosa-hcp-test
Display Name:               rosa-hcp-test
ID:                         <cluster_hcp_id>
External ID:                <cluster_hcp_id>
Control Plane:              ROSA Service Hosted
OpenShift Version:          4.Y.Z
Channel Group:              stable
DNS:                        <dns>

> AWS Account:                <AWS_id>
> AWS Billing Account:        <AWS_id>

Reviewer 2: I'll stop commenting on these because there are a lot more, but throughout the PR, change any `<AWS_id>` to lowercase `<aws_id>`, unless this is verbatim text from the actual output. I'm assuming though that this is just scrubbing the real AWS ID out for the sake of the example output.
```suggestion
AWS Account:                <aws_id>
AWS Billing Account:        <aws_id>
```

Author: replaced in 24 places in the 2 files, thanks

// * post_installation_configuration/preparing-for-users.adoc

:_mod-docs-content-type: PROCEDURE
[id="images-editing-image-registry-settings-hcp_{context}"]
= Editing image registry settings for {hcp-title}

You can change the image registry config with the `rosa edit` command.

[WARNING]
====
When the `allowedRegistries` parameter is defined, all registries, are blocked unless explicitly listed. To prevent pod failure, a list of Red Hat registries is automatically whitelisted, as they are required by payload images within your environment. The current list consists of `image-registry.openshift-image-registry.svc:5000,quay.io,registry.redhat.io` and it is also visible when running `rosa describe cluster`.
====

[NOTE]
====
You can change any registry related parameter which will trigger a rollout across all machinepools (all machinepool nodes will be recreated, following pod draining from each node).
====

.Procedure

* Update or edit the image registry for the cluster by running the following command:

+
[source,terminal]
----
$ rosa edit cluster --registry-config-insecure-registries <insecure_registries> \
   --registry-config-blocked-registries <insecure_registries> \

>    --registry-config-allowed-registries <allowed_registries> \

Reviewer 3: The two flags are mutually exclusive.We can just edit with one of them.

 --registry-config-blocked-registries <insecure_registries> \
   --registry-config-allowed-registries <allowed_registries> \

Author: removed "  --registry-config-blocked-registries <insecure_registries> \"

   --registry-config-allowed-registries-for-import <registry_name:insecure> \
   --registry-config-additional-trusted-ca <additional_trusted_ca_file>
----
+
.Example output
[source,terminal]
----
rosa edit cluster -c <cluster_name> --registry-config-blocked-registries <block_registries> \
? Changing any registry related parameter will trigger a rollout across all machinepools \
(all machinepool nodes will be recreated, following pod draining from each node). \
Do you want to proceed? Yes \

> I: Updated cluster '<cluster_name>'

Reviewer 3: It would be better to add description for this confirmation.

> ? Changing any registry related parameter will trigger a rollout across all machinepools \
> (all machinepool nodes will be recreated, following pod draining from each node). \
> Do you want to proceed? Yes \

Author: @yingzhanredhat  can you tell me what description you would like? please as I think the line is description.

Author: I am going to merge the PR and if a description needs to be added we can revisit after feature is released but @arendej and @andreadecorte gave me the go ahead to merge so will do that. @YiqinZhang I have applied your 2 other suggestions, thanks

 - arn:aws:iam::<AWS_id>:role/<operator_roles_prefix>-kube-system-kms-provider
 - arn:aws:iam::<AWS_id>:role/<operator_roles_prefix>-openshift-image-registry-installer-cloud-cred
 - arn:aws:iam::<AWS_id>:role/<operator_roles_prefix>-openshift-ingress-operator-cloud-credentials
 - arn:aws:iam::<AWS_id>:role/<operator_roles_prefix>-openshift-cluster-csi-drivers-ebs-cloud-credent
 - arn:aws:iam::<AWS_id>:role/<operator_roles_prefix>-openshift-cloud-network-config-controller-cloud
Managed Policies:           Yes
State:                      ready
Private:                    No
Delete Protection:          Disabled
Created:                    Oct 01 2030 09:48:52 UTC
User Workload Monitoring:   Enabled
OIDC Endpoint URL:          https://<endpoint> (Managed)
Audit Log Forwarding:       Disabled
External Authentication:    Disabled
Etcd Encryption:            Disabled
Registry Configuration:
 - Allowed Registries:      <allowed_registry> <1>
 - Insecure Registries:     <insecure_registry> <2>
 - Allowed Registries for Import: <3>
    - Domain Name:          <domain_name> <4>
    - Insecure:             true <5>
----

> <1> `Allowed Registries`: A comma-separated list of registries for which image pull and push actions are allowed.

Reviewer 3: In above edition, it used 'blocked registries',so it should be consistent.

Author: Updated line 29 for "allowed" instead of "blocked":
![image](https://github.com/user-attachments/assets/18de7a53-52a9-4817-97b7-5f8f251d3fa6)

// Module included in the following assemblies:
//
// * openshift_images/image-configuration-hcp.adoc
// * post_installation_configuration/preparing-for-users.adoc

:_mod-docs-content-type: PROCEDURE
[id="images-editing-image-registry-settings-hcp_{context}"]
= Editing image registry settings for ROSA with HCP

> Description needed

Reviewer 4: Update or edit the image registry for the cluster, you can run the following command:
